core/src/main/java/bisq/core/app/BisqSetup.java

```diff
@@ -482,14 +482,21 @@ private void maybeShowTac() {
         }
     }
 
-    private void detectLocalBitcoinNode(Runnable nextStep) {
-        BaseCurrencyNetwork baseCurrencyNetwork = config.baseCurrencyNetwork;
-        if (config.ignoreLocalBtcNode || baseCurrencyNetwork.isDaoRegTest() || baseCurrencyNetwork.isDaoTestNet()) {
+    private void maybeCheckLocalBitcoinNode(Runnable nextStep) {
+        if (localBitcoinNode.willIgnore()) {
             nextStep.run();
             return;
         }
 
-        localBitcoinNode.detectAndRun(nextStep);
+        // Here we only want to provide the user with a choice (in a popup) in case a
+        // local node is detected, but badly configured.
+        var detectedButMisconfigured = localBitcoinNode.isDetectedButMisconfigured();
+        if (detectedButMisconfigured) {
+            displayLocalNodeMisconfigurationHandler.accept(nextStep);
```

Comment:
```suggestion
            if (displayLocalNodeMisconfigurationHandler != null) {
                displayLocalNodeMisconfigurationHandler.accept(nextStep);
            }
```

Add a check to prevent possible NullPointerException.